sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryExecutionErrors.scala

```diff
@@ -593,6 +593,16 @@ object QueryExecutionErrors {
     new QueryExecutionException(message, e)
   }
 
+  def unsupportedParquetDictionaryDecodingError(
+      valueType: String,
+      dictionary: String,
```

Comment:
Just let it take the dictionary object here? you can avoid having callers all call .getClass().getSimpleName()

Reply:
> Just let it take the dictionary object here? you can avoid having callers all call .getClass().getSimpleName()

Hive tried this, but QueryExecutionError is under catalyst module, spark didn't add dependency of parquet in this module.

Reply:
AnyRef would work here but probably too odd to be worth it

Reply:
> AnyRef would work here but probably too odd to be worth it

Yea, but I am confused why QueryExecutionError under catalyst module. And such as `logicalHintOperatorNotRemovedDuringAnalysisError`, it's a error about analysis. seems a little mess

Reply:
Yes it's a bit strange. The modules are organized as follow:

- **Catalyst (sql/catalyst)** - An implementation-agnostic framework for manipulating trees of relational operators and expressions.
- **Execution (sql/core)** - A query planner / execution engine for translating Catalyst’s logical query plans into Spark RDDs. This component also includes a new public interface, SqlContext, that allows users to execute SQL or structured scala queries against existing RDDs and Parquet files. 
- **Hive Metastore Support (sql/hive)** - An extension of SqlContext called HiveContext that allows users to write queries using a subset of HiveQL and access data from a Hive Metastore using Hive SerDes. There are also wrappers that allows users to run queries that include Hive UDFs, UDAFs, and UDTFs.

So it seems `QueryExecutionError` are more suited for `sql/core`

Reply:
Expressions are in sql/catalyst and can throw `QueryExecutionError` as well. Maybe we should abstract expression evaluation and put the implementation in sql/core, but that's too much work.